### PR TITLE
Introduce ConfigMap and Secret

### PR DESCRIPTION
**Pull request by gnaponie**
Introduce ConfigMap and Secret, they still have placeholders values. Also include all the necessary information about the component in the PipelineRun.

### CODE REVIEW

**Comment by gnaponie** [comment]
I need to fix the tests.

**Review by FernandesMF — APPROVED** [review]
Looks good to me! I just have one minor question/suggestion.

**Review by FernandesMF — APPROVED** [review]

File: internal/controller/dependencyupdatecheck_controller.go [hunk 1]
```diff
@@ -55,45 +57,59 @@ func NewDependencyUpdateCheckReconciler(client client.Client, scheme *runtime.Sc
...
+	}
+}
+
 // createPipelineRun creates and returns a new PipelineRun
 func (r *DependencyUpdateCheckReconciler) createPipelineRun(comp component.GitComponent, ctx context.Context) (*tektonv1.PipelineRun, error) {
 
 	log := ctrllog.FromContext(ctx).WithName("DependencyUpdateCheckController")
 	ctx = ctrllog.IntoContext(ctx, log)
-	name := fmt.Sprintf("renovate-%d-%s-%s", comp.GetTimestamp(), RandomString(8), comp.GetName())
+	name := fmt.Sprintf("renovate-%d-%s-%s-%s", comp.GetTimestamp(), RandomString(8), comp.GetApplication(), comp.GetName())
```
Comment (qixiang) [inline internal/controller/dependencyupdatecheck_controller.go:94]: I'd suggest not adding application, some application and component names are long, and PipelineRun has a max chars limitation to 63 chars.

File: internal/controller/dependencyupdatecheck_controller.go [hunk 2]
```diff
@@ -55,45 +57,59 @@ func NewDependencyUpdateCheckReconciler(client client.Client, scheme *runtime.Sc
...
+
+	r.CreateConfigMap(comp, ctx, name)
+	r.CreateSecret(comp, ctx, name)
```
Comment (qixiang) [inline internal/controller/dependencyupdatecheck_controller.go:97]: As Mario pointed out, we need to create the ConfigMap/Secret first, not just the definition.

**Review by qixiang — COMMENTED** [review]
I made some slight changes to allow flexible attachment for the configmap and secrets, wdyt: https://github.com/konflux-ci/mintmaker/compare/refactor-config-map...qixiang:mintmaker:pipeline-configmap-secret

**Pull request merged**